oauth2_http/java/com/google/auth/oauth2/DefaultCredentialsProvider.java

```diff
@@ -282,6 +259,43 @@ private boolean runningOnAppEngine() {
         cause);
   }
 
+  final GoogleCredentials getFromGcloudCliWellKnownFile(HttpTransportFactory transportFactory)
+      throws IOException {
+    return tryGetFromWellKnownCredentialsFile(transportFactory);
+  }
+
+  @VisibleForTesting
+  final GoogleCredentials tryGetFromWellKnownCredentialsFile(HttpTransportFactory transportFactory)
```

Comment:
IIRC, we agreed on exposing the logic that returns the path, not the actual credential. This logic has some ADC-specific stuff, like printing warnings about user credentials, that you don't need.

Reply:
Seems like the getWellKnownCredentialsFile is what you need, or new method that encapsulates just the Path part of the File creation.

Reply:
I exposed a method to wrap `getWellKnownCredentialsFile`'s path.

Reply:
Yep, that is better

Reply:
You probably want to revert this change (as no longer needed) or cover it with tests... either works